refactor(bdk_hwi): remove `bdk_hwi`, as `HWISigner`'s being moved to `rust-hwi`

oleonardolima opened the pull request:
<!-- You can erase any parts of this template not applicable to your Pull Request. -->
fixes #1516
depends on https://github.com/bitcoindevkit/rust-hwi/pull/104

### Description

<!-- Describe the purpose of this PR, what's being adding and/or fixed -->

This PR removed the `bdk_hwi` crate, and the dangling `hardwaresigner.rs` file from `bdk_wallet` crate.

It simplifies the BDK project workspace as it removes the dependency on `rust-hwi`, pyo3 and some required steps on CI.

The removal is fine, as the `HWISigner` is being moved to `rust-hwi` project instead, please check: https://github.com/bitcoindevkit/rust-hwi/pull/104

### Notes to the reviewers

<!-- In this section you can include notes directed to the reviewers, like explaining why some parts
of the PR were done in a specific way -->

Please let me know what you think about the removal, and if I missed anything.

### Changelog notice

<!-- Notice the release manager should include in the release tag message changelog -->
<!-- See https://keepachangelog.com/en/1.0.0/ for examples -->

- Removed dangling unused `hardwaresigner.rs` file from `bdk_wallet` crate.
- Removed `bdk_hwi` crate, moved to `rust-hwi` project.

### Checklists

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

#### New Features:

* [ ] I've added tests for the new feature
* [ ] I've added docs for the new feature

#### Bugfixes:

* [ ] This pull request breaks the existing API
* [ ] I've added tests to reproduce the issue which are now passing
* [x] I'm linking the issue being fixed by this PR

oleonardolima committed:
refactor(wallet)!: remove dangling unused `hardwaresigner.rs` file

It seems this file was left out on the previous migration from hardware
signers from `bdk_wallet` crate to the `bdk_hwi`, but it should've been
removed.

oleonardolima committed:
refactor(bdk_hwi)!: remove `bdk_hwi`

- removes `bdk_hwi` crate, as `HWISigner`'s being moved to
rust-hwi.
- please refer to: https://github.com/bitcoindevkit/rust-hwi/pull/104

oleonardolima committed:
fix(ci)!: remove `Dockerfile.ledger` and `hwi` steps from coverage step

oleonardolima commented on crates/hwi/src/signer.rs:
```diff
@@ -1,94 +0,0 @@
-}
-
-// TODO: re-enable this once we have the `get_funded_wallet` test util
-// #[cfg(test)]
-// mod tests {
-//     #[test]
-//     fn test_hardware_signer() {
-//         use std::sync::Arc;
-//
-//         use bdk_wallet::tests::get_funded_wallet;
-//         use bdk_wallet::signer::SignerOrdering;
-//         use bdk_wallet::bitcoin::Network;
-//         use crate::HWISigner;
-//         use hwi::HWIClient;
-//
-//         let mut devices = HWIClient::enumerate().unwrap();
-//         if devices.is_empty() {
-//             panic!("No devices found!");
-//         }
-//         let device = devices.remove(0).unwrap();
-//         let client = HWIClient::get_client(&device, true, Network::Regtest.into()).unwrap();
-//         let descriptors = client.get_descriptors::<String>(None).unwrap();
-//         let custom_signer = HWISigner::from_device(&device, Network::Regtest.into()).unwrap();
-//
-//         let (mut wallet, _) = get_funded_wallet(&descriptors.internal[0]);
-//         wallet.add_signer(
-//             bdk_wallet::KeychainKind::External,
-//             SignerOrdering(200),
-//             Arc::new(custom_signer),
-//         );
-//
-//         let addr = wallet.get_address(bdk_wallet::AddressIndex::LastUnused);
-//         let mut builder = wallet.build_tx();
-//         builder.drain_to(addr.script_pubkey()).drain_wallet();
-//         let (mut psbt, _) = builder.finish().unwrap();
-//
-//         let finalized = wallet.sign(&mut psbt, Default::default()).unwrap();
-//         assert!(finalized);
-//     }
-// }
```
I don't think this a test we can move, at least not in its current state as it depends on `bdk_wallet::common::tests` module.

I'll need to rework it on the new `hwi::signer` to cover what some helper fns are doing manually.

notmandatory commented:
bitcoindevkit/rust-hwi#104 is merged.

notmandatory commented:
I rebased this PR and removed the `hwi/Cargo.toml` file which looks like it was missed.

oleonardolima committed:
chore: remove `ci/automation.json` file, used by Dockerfile.ledger

oleonardolima commented:
> I rebased this PR and removed the `hwi/Cargo.toml` file which looks like it was missed.

Thanks! I just noticed that I forgot to remove the `automation.json` file used by the simulator, I removed it in cd8ddfe18441c562342bcaccf7cae3a0d9ebbd92

notmandatory reviewed (approved):
ACK cd8ddfe18441c562342bcaccf7cae3a0d9ebbd92
